### PR TITLE
【gc】support archive and full mode

### PR DESCRIPTION
**Pull request by 516108736**

### CODE REVIEW

File: core/rootblockchain.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ const (
 	receiptsCacheLimit        = 32
 	maxFutureBlocks           = 32
 	maxTimeFutureBlocks       = 30
-	triesInMemory             = 256
+	triesInMemory             = 128
```
Comment (pingke) [inline core/rootblockchain.go]: still use 256?

File: core/minorblockchain_addon.go [hunk 2]
```diff
@@ -406,11 +407,54 @@ func (m *MinorBlockChain) InitFromRootBlock(rBlock *types.RootBlock) error {
...
+func (m *MinorBlockChain) reRunBlockWithState(block *types.MinorBlock) error {
+	blockWithoutState := make([]types.IBlock, 0)
+	for {
+		if _, err := m.StateAt(block.Meta().Root); err == nil {
+			log.Info("reRunBlockWithState blockchain to past state", "number", block.Number(), "hash", block.Hash().String())
+			break
+		}
+		blockWithoutState = append(blockWithoutState, block)
+		block = m.GetMinorBlock(block.ParentHash())
+		if qkcCommon.IsNil(block) {
```
Comment (pingke) [inline core/minorblockchain_addon.go]: check block high == 0 ?

File: core/minorblockchain.go [hunk 3]
```diff
@@ -282,9 +280,9 @@ func (m *MinorBlockChain) loadLastState() error {
 	if _, err := m.StateAt(currentBlock.GetMetaData().Root); err != nil {
 		// Dangling block without a state associated, init from scratch
 		log.Warn("Head state missing, repairing chain", "number", currentBlock.NumberU64(), "hash", currentBlock.Hash())
-		if err := m.repair(&currentBlock); err != nil {
-			return err
-		}
+		//if err := m.repair(&currentBlock); err != nil {
```
Comment (pingke) [inline core/minorblockchain.go]: why remove?

Comment (516108736) [reply core/minorblockchain.go]: will repair in InitFromRootBlock
will remove repair func

File: core/minorblockchain_addon.go (hunk 2 above)
Comment (516108736) [reply core/minorblockchain_addon.go]: always have trie while block height=0

**Pull request merged**